src/main/java/com/example/gistcompetitioncnserver/answer/AnswerService.java

```diff
@@ -1,56 +1,106 @@
-        answer.setCreated(LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
-        answer.setCategory(post.getCategory());
-        answer.setTitle("RE:"+post.getTitle());
-        return answerRepository.save(answer);
+    @Transactional(readOnly = true)
+    public Answer retrieveAnswerByPostId(Long postId){
+        checkExistenceByPostId(postId);
+        return findAnswerByPostId(postId);
     }
 
-    public List<Answer> retrieveAllAnswers(){
-        return answerRepository.findAll();
+    public Long getNumberOfAnswers(){
+        return answerRepository.count();
     }
 
-    public List<Answer> retrieveAnswersByUserId(Long user_id){
-        return answerRepository.findByUserId(user_id);
+    @Transactional
+    public void updateAnswer(Long updaterId, Long postId, AnswerRequestDto changeRequest) {
+        checkExistenceByPostId(postId);
+        Answer answer = findAnswerByPostId(postId);
+
+        User updater = findUserById(updaterId);
+        if (!canUpdate(updater, answer)) {
+            throw new CustomException("수정권한이 없는 user입니다.");
+        }
+        answer.updateContent(changeRequest.getContent());
     }
 
-    public Optional<Answer> retrieveAnswer(Long id){
-        return answerRepository.findById(id);
+    private boolean canUpdate(User user, Answer answer) {
+        Long answerOwner = answer.getUserId();
+        return answerOwner.equals(user.getId()) && user.getUserRole()==UserRole.MANAGER;
     }
 
-    public Long getPageNumber(){
-        return answerRepository.count();
+    @Transactional
+    public void deleteAnswer(Long eraserId, Long postId) {
+        checkExistenceByPostId(postId);
+        Answer answer = findAnswerByPostId(postId);
+
+        User eraser = findUserById(eraserId);
+        if (!canDelete(eraser, answer)) {
+            throw new CustomException("지울 수 있는 권한이 없습니다");
+        }
+        answerRepository.deleteByPostId(postId);
     }
 
-    public List<Answer> getAnswersByCategory(String categoryName){
-        return answerRepository.findByCategory(categoryName);
+    private boolean canDelete(User user, Answer answer) {
+        Long answerOwner = answer.getUserId();
+        return user.getUserRole()==UserRole.ADMIN || answerOwner.equals(user.getId()) && user.getUserRole()==UserRole.MANAGER;
     }
 
-    @Transactional
-    public void deleteAnswer(Long id){
-        answerRepository.deleteById(id);
+
+    private void checkExistenceByPostId(Long postId) {
+        if (!postRepository.existsById(postId)) {
+            throw new CustomException("존재하지 않는 post입니다");
+        }
     }
 
+    private User findUserById(Long userId) {
+        return userRepository.findById(userId).orElseThrow(()-> new CustomException("존재하지 않는 user입니다"));
+    }
 
+    private Answer findAnswerByPostId(Long postId) {
+        return answerRepository.findByPostId(postId).orElseThrow(
+                () -> new CustomException("해당 post에는 답변이 존재하지 않습니다.")
+        );
+    }
```

Comment:
```suggestion
    private Answer findAnswerByPostId(Long postId) {
        return answerRepository.findByPostId(postId)
                       .orElseThrow(() -> new CustomException("해당 post에는 답변이 존재하지 않습니다."));
    }
```

요 정도로 위와 형태를 맞추는 것도 좋을 것 같다는 생각입니다! (요건 개인 취향) 결정은 여러분의 생각대로!